### ISSUE TITLE
AttributeError: ‘torch.dtype’ object has no attribute 'type'

### ISSUE THREAD

**Issue by ybj-catherine** [Thinklab-SJTU/EDA-AI#10]
![cd4b3850dd99ff84533b2b4de75d35b](https://user-images.githubusercontent.com/90683795/167785752-3d7c0417-c1d7-4002-8f26-91544e325ecd.jpg)
when I executed the training on Macro placement, I got this error:
AttributeError: ‘torch.dtype’ object has no attribute 'type'
I refer to this solution: https://discuss.pytorch.org/t/attributeerror-torch-dtype-object-has-no-attribute-type/135871
but then I got this:
![5423496e36b63283635a8a16a1c0d3e](https://user-images.githubusercontent.com/90683795/167786798-6e08deb1-cd3d-4047-a0d3-8f65287aaaca.png)
I wonder whether editing code is needed or not 

Thanks

**Comment by Rogerity** [comment]
I have updated main.py, so the problem won't happen any more.

**Comment by ybj-catherine** [comment]
> I have updated main.py, so the problem won't happen any more.

actually I downloaded the code three days before, and I saw you updated the file on April 5th, but I still got this error

**Comment by Dinple** [comment]
Hi all, I found a way to work around this. 
Replace the print statement in line 194 inside `main.py` with the following:
`            print(
                "Updates {}, num timesteps {}, FPS {} \n Last {} training episodes: mean/median reward {:.1f}/{:.1f}, min/max reward {:.1f}/{:.1f}\n"
                .format(j, total_num_steps,
                        int(total_num_steps / (end - start)),
                        len(episode_rewards), torch.mean(torch.stack(list(episode_rewards))),
                        torch.median(torch.stack(list(episode_rewards))), torch.min(torch.stack(list(episode_rewards))),
                        torch.max(torch.stack(list(episode_rewards))), dist_entropy, value_loss,
                        action_loss))`

Basically, `episode_rewards` as a `collection.deque` object with a list of tensor objects cannot be input to NumPy directly. Here, I convert it to a list and then stack the value together. I changed all the functions to the corresponding torch functions. 

Please let me know if there are any mistakes. Hope this helps!

**Comment by ybj-catherine** [comment]
> Hi all, I found a way to work around this. Replace the print statement in line 194 inside `main.py` with the following: ` print( "Updates {}, num timesteps {}, FPS {} \n Last {} training episodes: mean/median reward {:.1f}/{:.1f}, min/max reward {:.1f}/{:.1f}\n" .format(j, total_num_steps, int(total_num_steps / (end - start)), len(episode_rewards), torch.mean(torch.stack(list(episode_rewards))), torch.median(torch.stack(list(episode_rewards))), torch.min(torch.stack(list(episode_rewards))), torch.max(torch.stack(list(episode_rewards))), dist_entropy, value_loss, action_loss))`
> 
> Basically, `episode_rewards` as a `collection.deque` object with a list of tensor objects cannot be input to NumPy directly. Here, I convert it to a list and then stack the value together. I changed all the functions to the corresponding torch functions.
> 
> Please let me know if there are any mistakes. Hope this helps!

Thanks for your solution! I have solved this problem by your method.
